addons/l10n_it_edi_sdicoop/models/account_invoice.py

```diff
@@ -22,3 +22,9 @@ def send_pec_mail(self):
         # With SdiCoop web-service, no need to send PEC mail.
         # Set the state to 'other' because the invoice should not be managed par l10n_it_edi.
         self.l10n_it_send_state = 'other'
+
+    @api.depends('restrict_mode_hash_table', 'state', 'l10n_it_edi_transaction')
+    def _compute_show_reset_to_draft_button(self):
+        for move in self:
+            super(AccountMove, move)._compute_show_reset_to_draft_button()
+            move.show_reset_to_draft_button = move.show_reset_to_draft_button and not move.l10n_it_edi_transaction
```

Comment:
feedback from @jco-odoo :smile: 
- the super() only needs to be called once, on the entire recordset.
- the write() only needs to be called once
```suggestion
        super()._compute_show_reset_to_draft_button()
        self.filtered(lambda move: move.show_reset_to_draft_button and not move.l10n_it_edi_transaction).write({
            'show_reset_to_draft_button': False
        })
```

Reply:
Ok, c'est une bonne optimisation